tests/experimental/pydantic/test_conversion.py

```diff
@@ -1247,3 +1247,31 @@ class Test:
 
     assert test.optional_list == [1, 2, 3]
     assert test.optional_str is None
+
+
+def test_exclude_unset_parameter_when_convert_strawberry_type_to_pydantic_type():
+    class User(BaseModel):
+        id: int
+        name: str
+        surname: Optional[str]
+
+    @strawberry.experimental.pydantic.type(model=User, all_fields=True)
+    class UserType:
+        pass
+
+    user = User(id=1, name="leffe")
+    user_type = UserType(id=1, name="leffe")
+    user_type_pydantic = user_type.to_pydantic()
```

Comment:
can we have a test where the `user_type = UserType(id=1, name="leffe", surname=None)`. And check that the None in the dict doesn't get removed?

Reply:
and another one where the `surname: Optional[str] = "default"`. And don't set the surname, so it uses the default.

Reply:
> can we have a test where the user_type = UserType(id=1, name="leffe", surname=None). And check that the None in the dict doesn't get removed?

You mean dict is `user_type.to_pydantic.dict()` ?  `not user_type.to_pydantic.dict(exclude_unset=True)` ?
If it's the former, It's testable and I'll add it.

Reply:
> > can we have a test where the user_type = UserType(id=1, name="leffe", surname=None). And check that the None in the dict doesn't get removed?
> 
> You mean dict is `user_type.to_pydantic.dict()` ? not `user_type.to_pydantic.dict(exclude_unset=True)` ? If it's the former, It's testable and I'll add it.

no, this

```
user_type = UserType(id=1, name="leffe", surname=None)
user_type.to_pydantic.dict(exclude_unset=True)
```
Because now you are setting surname=None explicitly. So that is not unset. It is set. so the None should still be in the dict

Reply:
Thank you, I understood.
In fact, It is a case that I was concerned about while patching, but the existing tests all worked well, so I did't worry about that.
But after seeing your comment, I think it should be considered. rigth?